app/controllers/application_controller.rb

```diff
@@ -144,7 +144,7 @@ def authenticate_token
 
   # https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Products/SSO/CookieSpecs-20180906.docx
   def set_sso_cookie!
-    return unless Settings.sso.cookie_enabled
+    return unless Settings.sso.cookie_enabled && @session
```

Comment:
Ideally we wouldn't need this or we could avoid getting here with a nil session but I can't think through a use case that does that -- so I resorted to adding to this guard condition -- let me know if you see a better way to avoid the sentry error we are getting though :)

Reply:
According to the stack trace from the Sentry error, `set_sso_cookie!` was called from `saml_callback` in the `SessionsController` after it set `@session = sso_service.new_session`. It seems that the `initialize` method of `SSOService` never sets `new_session` if the SAML response is invalid. Perhaps that's where we should be handling the nil session.

Reply:
Or maybe only do the `after_login_actions` and redirect in `saml_callback` if a valid session came back from `SSOService`.

Reply:
Thanks for the feedback, @U-DON ! I think I'm making too many assumptions about the `@sso_service.persist_authentication!` method -- there is a conditional in the `saml_callback` that relies on it but we must be reaching a state where that's truthy but session is nil .. I'm going to update this with another approach that pulls in your suggestions -- let me know what you think about that approach when you have a chance.